plugin/trino-kinesis/src/main/java/io/trino/plugin/kinesis/KinesisRecordSet.java

```diff
@@ -290,6 +300,32 @@ private void getKinesisRecords()
             messagesRead += kinesisRecords.size();
         }
 
+        private Optional<Map<DecoderColumnHandle, FieldValueProvider>> decodeMessage(byte[] messageData)
+        {
+            if (!split.getCompressionCodec().isPresent()) {
+                return messageDecoder.decodeRow(messageData);
+            }
+            if (split.getCompressionCodec().get() == GZIP || split.getCompressionCodec().get() == AUTOMATIC) {
+                if (isGZipped(messageData)) {
+                    try (
+                            ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream(messageData);
+                            GZIPInputStream gZIPInputStream = new GZIPInputStream(byteArrayInputStream)) {
+                        return messageDecoder.decodeRow(gZIPInputStream.readAllBytes());
+                    }
+                    catch (IOException e) {
+                        throw new TrinoException(GENERIC_INTERNAL_ERROR, e);
+                    }
+                }
+                if (split.getCompressionCodec().get() == GZIP) {
+                    throw new TrinoException(GENERIC_INTERNAL_ERROR, format("Failed to decompress %s message", split.getCompressionCodec().get()));
+                }
+            }
+            if (split.getCompressionCodec().get() == AUTOMATIC) {
```

Comment:
We could add `UNCOMPRESSED` (default value) then we could merge this block with the one at top of this method.